gax-httpjson/src/main/java/com/google/api/gax/httpjson/HttpJsonCallSettings.java

```diff
@@ -29,39 +29,50 @@
  */
 package com.google.api.gax.httpjson;
 
-import com.google.api.core.BetaApi;
-import com.google.api.gax.rpc.ApiCallContext;
-import com.google.api.gax.rpc.ApiCallContextEnhancer;
-import com.google.api.gax.rpc.TransportChannel;
-import com.google.api.gax.rpc.TransportDescriptor;
-import com.google.auth.Credentials;
+/** Grpc-specific settings for creating callables. */
+public class HttpJsonCallSettings<RequestT, ResponseT> {
+  private final ApiMethodDescriptor<RequestT, ResponseT> methodDescriptor;
 
-/** Implementation of TransportDescriptor for http/json. */
-@BetaApi
-public class HttpJsonTransportDescriptor extends TransportDescriptor {
-  private HttpJsonTransportDescriptor() {}
+  private HttpJsonCallSettings(ApiMethodDescriptor<RequestT, ResponseT> methodDescriptor) {
+    this.methodDescriptor = methodDescriptor;
+  }
 
-  public static HttpJsonTransportDescriptor create() {
-    return new HttpJsonTransportDescriptor();
+  public ApiMethodDescriptor<RequestT, ResponseT> getMethodDescriptor() {
+    return methodDescriptor;
   }
 
-  @Override
-  public ApiCallContext createDefaultCallContext() {
-    return HttpJsonCallContext.createDefault();
+  public static <RequestT, ResponseT> Builder<RequestT, ResponseT> newBuilder() {
+    return new Builder<>();
   }
 
-  @Override
-  public ApiCallContext getCallContextWithDefault(ApiCallContext inputContext) {
-    return HttpJsonCallContext.getAsHttpJsonCallContextWithDefault(inputContext);
+  public static <RequestT, ResponseT> HttpJsonCallSettings<RequestT, ResponseT> of(
+      ApiMethodDescriptor<RequestT, ResponseT> methodDescriptor) {
+    return HttpJsonCallSettings.<RequestT, ResponseT>newBuilder()
+        .setMethodDescriptor(methodDescriptor)
+        .build();
   }
 
-  @Override
-  public ApiCallContextEnhancer getAuthCallContextEnhancer(Credentials credentials) {
-    return new HttpJsonAuthCallContextEnhancer(credentials);
+  public Builder toBuilder() {
+    return new Builder<>(this);
   }
 
-  @Override
-  public ApiCallContextEnhancer getChannelCallContextEnhancer(TransportChannel channel) {
-    return new HttpJsonChannelCallContextEnhancer(channel);
+  public static class Builder<RequestT, ResponseT> {
```

Comment:
Using `AutoValue` and using builders are independent decisions, since you can use `AutoValue.Builder`. 

This class in particular is one I see expanding, so I decided on a builder. 

I'm fine with adding easy construction methods in various classes in the future, but it's not something I'm going to worry about before GAX GA because such a thing is additive.